src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CartHandler.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace StoreKeeper\WooCommerce\B2C\Frontend\Handlers;
+
+use StoreKeeper\WooCommerce\B2C\I18N;
+use StoreKeeper\WooCommerce\B2C\Imports\ProductImport;
+use WooCommerce;
+
+class CartHandler
+{
+    public function addEmballageFee()
+    {
+        /* @var WooCommerce $woocommerce */
+        global $woocommerce;
+
+        $items = $woocommerce->cart->get_cart();
+
+        $totalEmballagePrice = 0.00;
+        foreach ($items as $values) {
+            $product = wc_get_product($values['product_id']);
+            if ($product) {
+                $emballagePrice = $product->get_meta(ProductImport::PRODUCT_EMBALLAGE_PRICE_META_KEY);
+                if ($emballagePrice) {
+                    $totalEmballagePrice += (float) $emballagePrice;
+                }
+            }
+        }
+
+        if ($totalEmballagePrice > 0) {
+            $woocommerce->cart->add_fee(__('Emballage fee', I18N::DOMAIN), $totalEmballagePrice);
```

Comment:
does it get removed if the you remove the product on cart? or only calculated on checkout

Reply:
Yes, fees added using `add_fee` does not persist until order is created on checkout.